src/Symfony/Component/Form/Extension/Core/DataTransformer/IntegerToLocalizedStringTransformer.php

```diff
@@ -40,6 +42,12 @@ public function __construct($scale = 0, $grouping = false, $roundingMode = self:
      */
     public function reverseTransform($value)
     {
+        $decimalSeparator = $this->getNumberFormatter()->getSymbol(\NumberFormatter::DECIMAL_SEPARATOR_SYMBOL);
+
+        if (\is_string($value) && false !== strpos($value, $decimalSeparator)) {
```

Comment:
The transformation is invoked in reverse order in `DateIntervalType` when widget = integer, also in the model layer, so it could receive a real floating value here, isn't it?

Reply:
Can you give an example? I am not sure that I understand what you mean.

Reply:
https://github.com/symfony/symfony/blob/5a3e8942032c1a49496fe8db61828707b45442f9/src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php#L116-L122

Reply:
Unlike `IntegerType` this transformation is invoked for model layer in reversed order, so I guess we should deal with real float values here.

Reply:
Basically, this is what I suggest:
```php
if (false !== strpos((string) $value, $decimalSeparator)) {
```

Reply:
Although, technically that should never happen because `DateIntervalToArrayTransformer`, so ignore my suggestion.